### DISCUSSION TITLE
A Little Help with simple 4 button Midi Controller

### DISCUSSION THREAD

**Discussion by Celticpure** [alf45tar/PedalinoMini#404] [Q&A]
HI everyone.

I am struggling a bit with setting up a simple four button midi controller footswitch for my Boss Katana amp through the Katana Librarian app for android. 

I have wiring completed and file system built and uploaded successfully and the device is running and connected to my WIFI network etc. So on this front I feel I have everything covered. It is the execution of the command setup that I am struggling with. 

I have 4 patches that I can assign and access through ControlChange commands via the four designated button/switches on my footswitch, However I want to use the a long button press (or similar command) to access the effects on on the Katana librarian app - i.e. Booster/Delay/Reverb and the three sub-setting on each with the same 4 buttons being also used for patch selection. 

Any help on how to go about this would be most welcome... photo below shows the footswitch. 

 

![IMG_20221128_114349](https://user-images.githubusercontent.com/96823448/204270781-7fcd83c5-76c9-4e42-90c9-c286c33658ef.jpg)

**Comment by alf45tar** [comment]
@Celticpure did you used momentary or latch switches?
Long Press cannot be used with latches for physical reason.

Assuming you configured pedals from 1 to 4, I can provide you with a configuration file as example.

**Reply by Celticpure** [reply]
I am using momentary, yes I have configured my installation for pedals 1-4 in the pedals.h file. 

Thank you that would be great. 👍

**Comment by alf45tar** [comment]
Unzip it before upload to PedalinoMini

[celticpure.cfg.zip](https://github.com/alf45tar/PedalinoMini/files/10107278/celticpure.cfg.zip)

**Reply by Celticpure** [reply]
Thank you very much. I will try this tomorrow. 👍

**Comment by Celticpure** [comment]
Thank you @alf45tar for the config file. I am unsure what the only option of 'OSC Message' is for? This is the only option I can select from the 'ON' drop down menu. I have attached a screenshot below. 


![Screenshot 2022-11-29 084751](https://user-images.githubusercontent.com/96823448/204482356-1013f4bf-43db-4afa-a473-1e1e271b27f6.png)

**Reply by alf45tar** [reply]
Did you imported Controls?
The configuration is for version 3. You are using version 2.

**Reply by Celticpure** [reply]
> Did you imported Controls?
> The configuration is for version 3. You are using version 2.

Yes I am using version 2.5.2. Did not import the file in the MACOSX folder as I assume it for Mac OS etc. I am on windows. Below are the files in the zip you attached. Thank you. 


 ![Screenshot_2022-11-29-09-41-08-802_com.cxinventor.file.explorer.jpg](https://user-images.githubusercontent.com/96823448/204494179-7247d664-e51f-40d1-b2ae-4eeafa631599.jpg)![Screenshot_2022-11-29-09-41-15-106_com.cxinventor.file.explorer.jpg](https://user-images.githubusercontent.com/96823448/204494207-b1c0a8c4-8028-4b49-acd8-62f6a1ad63a5.jpg)

**Reply by alf45tar** [reply]
Use the file celticpure.cfg outside _MACOSX folder. It is 3.83 kb in size.

**Reply by Celticpure** [reply]
> Use the file celticpure.cfg outside _MACOSX folder. It is 3.83 kb in size. 

Yes, that is the file I used. The only options I have is as shown in my screenshot above.

**Reply by alf45tar** [reply]
Here attached the file for version 2

[celticpure2.cfg.zip](https://github.com/alf45tar/PedalinoMini/files/10116738/celticpure2.cfg.zip)

**Reply by Celticpure** [reply]
Just a few photos of my midi pedal for Katana Librarian. Getting grips with Pedalino, a great project and very worthwhile sponsoring!!

[IMG_20221130_161158.jpg](https://user-images.githubusercontent.com/96823448/205149494-b03e8121-7897-4f9c-bfc0-3884c6615d4d.jpg)

![IMG_20221130_153800.jpg](https://user-images.githubusercontent.com/96823448/205149518-54354b4c-7317-4f55-b9af-28a14534420e.jpg)

![IMG_20221130_150417.jpg](https://user-images.githubusercontent.com/96823448/205149544-f7480399-5b0f-4036-97e6-ddcc19e865be.jpg)

![IMG_20221130_150410.jpg](https://user-images.githubusercontent.com/96823448/205149565-6311a23a-2be7-4579-8790-e0c4de3f14b9.jpg)

**Comment by rigr** [comment]
Nice!
